pkg/fileservice/file_service.go

```diff
@@ -103,23 +103,20 @@ type IOEntry struct {
 	// if number of bytes is unknown, set Size field to -1
 	ReaderForWrite io.Reader
 
-	// when reading, if the ToObject field is not nil, the returning object will be set to this field
-	// caches may choose to cache this object instead of caching []byte
-	// Data, WriterForRead, ReadCloserForRead may be empty if Object is not null
-	// if ToObject is provided, caller should always read Object instead of Data, WriterForRead or ReadCloserForRead
-	Object any
+	// When reading, if the ToObjectBytes field is not nil, the returning object's byte slice will be set to this field
+	// Data, WriterForRead, ReadCloserForRead may be empty if ObjectBytes is not null
+	// if ToObjectBytes is provided, caller should always read ObjectBytes instead of Data, WriterForRead or ReadCloserForRead
+	ObjectBytes []byte
```

Comment:
Do we really need this []byte?   Can we just reuse the Data field?

Reply:
We use 
- **Data** to hold the Compressed Data.
- **ObjectBytes** to hold the Uncompressed Data.

Combining both of them seemed to be difficult. Especially for this part:

```go
func (e *IOEntry) setObjectBytesFromData() error {
	if e.ToObjectBytes == nil {
		return nil
	}
	if len(e.Data) == 0 {
		return nil
	}
	obj, size, err := e.ToObjectBytes(bytes.NewReader(e.Data), e.Data)
	if err != nil {
		return err
	}
	e.ObjectBytes = obj
	e.ObjectSize = size
	return nil
}
```

I will see if that can be combined.

Reply:
> 

It's already zero-copy. Objects will not be copied if the cache is hit.
I think `[]byte` is less type-safe than `any`. And casting `[]byte` to other types is not faster than type assertion.

Reply:
It cannot be combined.